confutils.nim

```diff
@@ -220,6 +242,61 @@ proc parseCmdArgAux(T: type, s: TaintedString): T = # {.raises: [ValueError].} =
   mixin parseCmdArg
   parseCmdArg(T, s)
 
+proc completeCmdArg(T: type enum, val: TaintedString): seq[string] =
+  for e in low(T)..high(T):
+    let as_str = $e
+    if startsWithIgnoreStyle(as_str, val):
+      result.add($e)
+
+proc completeCmdArg(T: type SomeNumber, val: TaintedString): seq[string] =
+  return @[]
+
+proc completeCmdArg(T: type bool, val: TaintedString): seq[string] =
+  return @[]
+
+proc completeCmdArg(T: type string, val: TaintedString): seq[string] =
+  return @[]
+
+proc completeCmdArg*(T: type[InputFile|InputDir], val: TaintedString): seq[string] =
+  let (dir, name, ext) = splitFile(val)
+  let tail = name & ext
+  # Expand the directory component for the directory walker routine
+  let dir_path = if dir == "": "." else: expandTilde(dir)
+  # Dotfiles are hidden unless the user entered a dot as prefix
+  let show_dotfiles = len(name) > 0 and name[0] == '.'
+
+  for kind, path in walkDir(dir_path, relative=true):
+    if not show_dotfiles and path[0] == '.':
+      continue
+
+    # Do not show files if asked for directories, on the other hand we must show
+    # directories even if a file is requested to allow the user to select a file
+    # inside those
+    if type(T) is InputDir and kind notin {pcDir, pcLinkToDir}:
+      continue
+
+    # Note, no normalization is needed here
+    if path.startsWith(tail):
+      let match = if kind in {pcDir, pcLinkToDir}:
+        # Add a trailing slash so that completions can be chained
+        dir_path & DirSep & path & DirSep
+      else:
+        dir_path & DirSep & path
+
+      result.add(shellPathEscape(match))
+
+proc completeCmdArg*(T: type[OutFile|OutDir|OutPath], val: TaintedString): seq[string] =
+  return @[]
```

Comment:
It's still possible to complete existing directory names here

Reply:
Makes sense, will add it.